codec/codecgen/gen.go

```diff
@@ -180,6 +180,9 @@ func Generate(outfile, buildTag, codecPkgPath string, uid int64, useUnsafe bool,
 						//   FuncType, InterfaceType, StarExpr (ptr), etc
 						switch td.Type.(type) {
 						case *ast.StructType, *ast.Ident, *ast.MapType, *ast.ArrayType, *ast.ChanType:
+							if notRegexName.FindStringIndex(td.Name.Name) != nil {
```

Comment:
Can you combine both of these into one line i.e. if notRegexName... == nil && regexName... != nil { __add_to_types __ }

Thanks.